### ISSUE TITLE
when exclude __init__ throws AssertionError

### ISSUE THREAD

**Issue by kanchansrivastava** [sphinx-doc/sphinx#3177]
If I am trying to exclude __init__.py, it is throwing AssertionError.
I executed following command:

```
_sphinx-apidoc -o source/plugin ../plugins/ ../plugins/echo.py ../plugins/__init__.py -f_
error thrown:
Traceback (most recent call last):
  File "/Users/srivastk/virtualenv/proton/bin/sphinx-apidoc", line 11, in <module>
    sys.exit(main())
  File "/Users/srivastk/virtualenv/proton/lib/python3.5/site-packages/sphinx/apidoc.py", line 343, in main
    modules = recurse_tree(rootpath, excludes, opts)
  File "/Users/srivastk/virtualenv/proton/lib/python3.5/site-packages/sphinx/apidoc.py", line 230, in recurse_tree
    assert root == rootpath and root_package is None
AssertionError
```

**Comment by shimizukawa** [comment]
Thanks for your reporting.
I confirmed the problem. I'll take a look.

**Comment by shimizukawa** [comment]
@kanchan1991 I just fix the AssertionError (my local).
Now I have to decide a behavior if `__init__.py` file is excluded.

This is a result of apidoc without excluding `__init__.py`. This foo.rst file that is generated from foo package contains "Module contents" section.
```
foo package
===========

Submodules
----------

foo.bar module
--------------

.. automodule:: foo.bar
    :members:
    :undoc-members:
    :show-inheritance:


Module contents
---------------

.. automodule:: foo
    :members:
    :undoc-members:
    :show-inheritance:
```

I think if you exclude `foo/__init__.py`, this "Module contents" section should be removed from foo.rst. How do you think @kanchan1991 ?

**Comment by kanchansrivastava** [comment]
actually, there is no way to exclude "Module contents". I thought excluding foo/__init__.py can be a hack to solve my problem.

**Comment by tk0miya** [comment]
ping @shimizukawa 
Is this resolved? Or there is anything to do yet?

**Comment by takuan-osho** [comment]
I cloned https://github.com/pcav/faunalia-website, add `html_last_updated_fmt = '%d %b %Y'` and `language = 'fr'` in conf.py, and `sphinx-build -M html . build -P`. But no exception occurred.

The versions of sphinx and extension are here:

```
Sphinx==1.3.4
sphinx-bootstrap-theme==0.6.0
```

In the case of sphinx `1.6.5`, there is no exception occurred too.

**Comment by micahjsmith** [comment]
Not solved for me with `sphinx==1.8.0`

**Comment by tk0miya** [comment]
Please use `--implicit-namespaces`  option to let apidoc know the directory is a package.
Then the "Module contents" section will not be generated.

Thanks,
